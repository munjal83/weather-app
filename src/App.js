import React from "react";
import TorontoWeather from "../src/components/TorontoWeather";
import MontrealWeather from "../src/components/MontrealWeather";
import PhotoGallery from "../src/components/PhotoGallery";
import PhotoList from "./components/PhotoList";
import Contact from "./components/Contact";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import '../src/assets/css/App.css'

function getWeatherToronto() {
  return <TorontoWeather />;
}

function getWeatherMontreal() {
  return <MontrealWeather />
}

function getGallery() {
  return <PhotoGallery />
}

function getPhotoList() {
  return <PhotoList />;
}

function getContactForm() {
  return <Contact />;
}

function AppRouter() {
  return (
    <Router>
      <div>
        <nav>
          <ul>
            <li>
              <Link to="/weather/toronto">Toronto Weather</Link>
            </li>
            <li>
              <Link to="/weather/montreal">Montreal Weather</Link>
            </li>
            <li>
              <Link to="/gallery">Photo Gallery</Link>
            </li>
            <li>
              <Link to="/api/photo">Photo List</Link>
            </li>
            <li>
              <Link to="/contact">Contact Us</Link>
            </li>
          </ul>
        </nav>

        <Route path="/weather/toronto" component={getWeatherToronto} />
        <Route path="/weather/montreal" component={getWeatherMontreal} />
        <Route path="/gallery" component={getGallery} />
        <Route path="/api/photo" component={getPhotoList} />
        <Route path="/contact" component={getContactForm} />
      </div>
    </Router>
  );
}

export default AppRouter;
