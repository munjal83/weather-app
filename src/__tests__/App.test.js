import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import App from '../App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App Component Snapshot',()=>{
  test('testing app component', () => {
  let tree = create(<App />)
  expect(tree.toJSON()).toMatchSnapshot();
})
});