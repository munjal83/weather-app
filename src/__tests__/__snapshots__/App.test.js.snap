// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot testing app component 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="/weather/toronto"
          onClick={[Function]}
        >
          Toronto Weather
        </a>
      </li>
      <li>
        <a
          href="/weather/montreal"
          onClick={[Function]}
        >
          Montreal Weather
        </a>
      </li>
      <li>
        <a
          href="/gallery"
          onClick={[Function]}
        >
          Photo Gallery
        </a>
      </li>
      <li>
        <a
          href="/api/photo"
          onClick={[Function]}
        >
          Photo List
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact Us
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
